linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarGen.scala

```diff
@@ -134,64 +88,13 @@ private[emitter] final class VarGen(jsGen: JSGen, nameGen: NameGen,
       implicit pos: Position): Tree = {
     typeRef match {
       case primRef: PrimRef =>
-        VarRef(coreJSLibVarIdent(field, primRef))
+        classVar(field, primRef)
 
       case ClassRef(className) =>
         classVar(field, className)
     }
   }
 
-  def coreJSLibVar(field: String)(implicit pos: Position): Tree =
-    VarRef(coreJSLibVarIdent(field))
-
-  def coreJSLibClassDef(name: String, parentClass: Option[Tree],
-      members: List[Tree])(implicit pos: Position): Tree = {
-    ClassDef(Some(coreJSLibVarIdent(name)), parentClass, members)
-  }
-
-  def coreJSLibFunctionDef(name: String, args: List[ParamDef],
-      body: Tree)(implicit pos: Position): Tree = {
-    FunctionDef(coreJSLibVarIdent(name), args, body)
-  }
-
-  def coreJSLibFunctionDef(name: String, primRef: PrimRef,
-      args: List[ParamDef], body: Tree)(implicit pos: Position): Tree = {
-    FunctionDef(coreJSLibVarIdent(name, primRef), args, body)
-  }
-
-  def coreJSLibVarDecl(name: String)(implicit pos: Position): Tree =
-    genEmptyMutableLet(coreJSLibVarIdent(name))
-
-  def coreJSLibVarDef(name: String, rhs: Tree)(implicit pos: Position): Tree =
-    genConst(coreJSLibVarIdent(name), rhs)
-
-  def coreJSLibVarDef(name: String, primRef: PrimRef, rhs: Tree)(
-      implicit pos: Position): Tree = {
-    genConst(coreJSLibVarIdent(name, primRef), rhs)
-  }
-
-  private def coreJSLibVarIdent(field: String)(implicit pos: Position): Ident =
-    genericIdent(field)
-
-  private def coreJSLibVarIdent(field: String, primRef: PrimRef)(
-      implicit pos: Position): Ident = {
-    // The mapping in this function is an implementation detail of the emitter
-    val subField = primRef.tpe match {
-      case NoType      => "V"
-      case BooleanType => "Z"
-      case CharType    => "C"
-      case ByteType    => "B"
-      case ShortType   => "S"
-      case IntType     => "I"
-      case LongType    => "J"
-      case FloatType   => "F"
-      case DoubleType  => "D"
-      case NullType    => "N"
-      case NothingType => "E"
-    }
-    genericIdent(field, subField)
-  }
-
   def fileLevelVar(field: String, subField: String,
```

Comment:
Why didn't `fileLevelVar` receive the same `Scope`-based treatment?

Reply:
Because for `fileLevelVar`, the scope determination is at the call site (`subField` is merely a string). For modules, this becomes even more relevant, since `Scope` needs to provide a `reprClass` to determine the module the field is in. For `fileLevelVar`s we cannot determine this module.

We could introduce different types of `Scope` for this, but it seems overkill, since `VarGen` would still have to offer different methods for this.

Reply:
Ah right, that makes sense. And yes, given that, probably `globalVar` is the most appropriate name.